File: app/src/main/java/com/aaronbrecher/neverlate/viewmodels/BaseViewModel.java
package com.aaronbrecher.neverlate.viewmodels;

import android.app.Application;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import com.aaronbrecher.neverlate.AppExecutors;
import com.aaronbrecher.neverlate.database.EventsRepository;
import com.aaronbrecher.neverlate.models.Event;

public abstract class BaseViewModel extends ViewModel{
    protected Application mApplication;
    protected EventsRepository mEventsRepository;
    protected AppExecutors mAppExecutors;
    private MutableLiveData<Event> mEvent;

    BaseViewModel(EventsRepository eventsRepository, Application application, AppExecutors appExecutors){
        this.mEventsRepository = eventsRepository;
        this.mApplication = application;
        this.mAppExecutors = appExecutors;
    }

    public MutableLiveData<Event> getEvent() {
        if (mEvent == null)
            mEvent = new MutableLiveData<>();
        return mEvent;
    }

    public void setEvent(Event event) {
        if (mEvent == null) {
            mEvent = new MutableLiveData<>();
        }
        mEvent.postValue(event);
    }
}

File: app/src/main/java/com/aaronbrecher/neverlate/viewmodels/BaseViewModel.kt
package com.aaronbrecher.neverlate.viewmodels

import android.app.Application
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

import com.aaronbrecher.neverlate.AppExecutors
import com.aaronbrecher.neverlate.database.EventsRepository
import com.aaronbrecher.neverlate.models.Event

abstract class BaseViewModel internal constructor(protected var mEventsRepository: EventsRepository,
                                                  protected var mApplication: Application?,
                                                  protected var mAppExecutors: AppExecutors) : ViewModel() {

    var mEvent: MutableLiveData<Event> = MutableLiveData()

    open fun setEvent(event: Event) {
        mEvent.postValue(event)
    }
}
